src/prefect/transactions.py

```diff
@@ -243,12 +243,12 @@ def rollback(self) -> bool:
         try:
             for hook in reversed(self.on_rollback_hooks):
                 hook_name = _get_hook_name(hook)
-                if self.logger:
+                if hook_name != "__prefect_task_rollback_handler" and self.logger:
```

Comment:
this is pretty hacky. 😭 

considered refactoring hooks here to take an object to hold metadata like skip_logging or something rather than rely on a magic function name, but I wasn't convinced the complexity was worth it.. 

This is purely so that in the majority use-case where a user is not defining any rollback hooks for a task, this "system/prefect-built-in" hook doesn't surface in logs in the same way that user defined hooks would to reduce "system-level" noise.

Reply:
f10ecef changed this to make the logging behavior of the task run engine rather than the transaction. This is a little more inline with how the task run engine is responsible for similar logging of other hooks (see `TaskRunEngine.call_hooks`). This change also made it easier to only wrap user-defined hooks with the logging and keep the new hook that updates the task's state transparent to users.